backends/dpdk/dpdkArch.h

```diff
@@ -1320,5 +1320,91 @@ class CollectProgramStructure : public PassManager {
         }));
     }
 };
+
+/* Helper class to detect use of IPSec accelerator */
+class IsIPSecUsed : public Inspector {
+    bool &is_ipsec_used;
+    int &sa_id_width;
+    P4::ReferenceMap *refMap;
+    P4::TypeMap *typeMap;
+
+ public:
+    IsIPSecUsed(bool &is_ipsec_used, int &sa_id_width, P4::ReferenceMap *refMap,
+                P4::TypeMap *typeMap)
+        : is_ipsec_used(is_ipsec_used),
+          sa_id_width(sa_id_width),
+          refMap(refMap),
+          typeMap(typeMap) {}
+    bool preorder(const IR::MethodCallStatement *mcs) override {
+        auto mi = P4::MethodInstance::resolve(mcs->methodCall, refMap, typeMap);
+        if (auto a = mi->to<P4::ExternMethod>()) {
+            if (a->originalExternType->getName().name == "ipsec_accelerator") {
+                if (a->method->getName().name == "enable") {
+                    is_ipsec_used = true;
+                } else if (a->method->getName().name == "set_sa_index") {
```

Comment:
why are doing this here, if goal of this pass is to check wether ipsec used or not?

Reply:
I will change the pass name to CollectIPSecInfo. This can be used to collect all information required like whether ipsed is used or not, what is the bitwidth for sa_id and ip_hdr_offset when we implement set_ip_header_offset.